packages/@react-aria/selection/src/useSelectableCollection.ts

```diff
@@ -268,7 +268,9 @@ export function useSelectableCollection(options: AriaSelectableCollectionOptions
           // in the collection, so that the browser default behavior will apply starting from that element
           // rather than the currently focused one.
           if (e.shiftKey) {
-            ref.current.focus();
+            // Needs flushSync to make sure expandedKeys is updated before focus is moved out of the containing Menu's FocusScope. Without this
+            // the menu's contain isn't adjusted in time resulting in FocusScope contain keydown handlers not being cleaned up. Caught in test, doesn't happen in browser cuz lucky
+            flushSync(() => ref.current.focus());
```

Comment:
Ah we actually don't need this anymore since we let focus move to the preceeding/next element adjacent to the menu button rather than shifting focus between the submenutrigger and its submenu, I'll remove